### PR TITLE
Split InitializeKeycloakConfig and add KeycloakContainerTest

### PR DESCRIPTION
**Pull request by lmsurpre**
1. Introduce a dependency on the Keycloak testcontainers project
  * Note: this depends on a yet-to-be-released SNAPSHOT version of
https://github.com/dasniko/testcontainers-keycloak
2. Split InitializeKeycloakConfig into a Main and a Configurator; this
should make it easier to use the configurator from the new test
3. Introduce a placeholder test that
  * Uses the testcontainers-keycloak project to launch a keycloak
container from the JUnit
  * Uses the keycloak-config project to configure the container with a
test tenant
  * Retrieves the server info
4. Add the `jboss-deployment-structure.xml` back in. I wasn't sure if it
was needed or not but without it I'm seeing
"java.lang.NoClassDefFoundError: Failed to link" errors while
initializing the test realm with a PatientPrefixUserAttributeMapper.

### CODE REVIEW

File: keycloak-config/src/main/java/org/alvearie/keycloak/config/Main.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+/**
+ * Reads the KeycloakConfig and applies it to the target Keycloak
+ *
+ * Command line arguments:
+ *   configFile: the file path to a JSON file containing Keycloak configuration properties
+ */
+public class Main {
+
+    private static final String CONFIG_FILE_PATH_OPTION = "configFile";
+    private static final String CLASS_NAME = "InitializeKeycloakConfig";
```
Comment (tbieste) [inline keycloak-config/src/main/java/org/alvearie/keycloak/config/Main.java:30]: Would be good to have the CLASS_NAME and actual class name be the same.

**Review by tbieste — APPROVED** [review]
1 minor comment; otherwise LGTM

File: keycloak-config/src/main/java/org/alvearie/keycloak/config/Main.java (hunk 1 above)
Comment (lmsurpre) [reply keycloak-config/src/main/java/org/alvearie/keycloak/config/Main.java:30]:
```suggestion
    private static final String CLASS_NAME = "Main";
```

File: keycloak-extensions/pom.xml [hunk 2]
```diff
@@ -32,13 +33,49 @@
...
+        <dependency>
+            <groupId>org.testcontainers</groupId>
+            <artifactId>testcontainers</artifactId>
+            <version>${testcontainers.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.github.dasniko</groupId>
+            <artifactId>testcontainers-keycloak</artifactId>
+            <version>1.5.1-SNAPSHOT</version>
```
Comment (lmsurpre) [inline keycloak-extensions/pom.xml:64]:
```suggestion
            <version>1.6.0</version>
```

**Pull request merged**